azkaban-common/src/main/java/azkaban/user/UserUtils.java

```diff
@@ -78,16 +82,18 @@ static void setupWatch(final Map<String, ParseConfigFile> configFileMap) throws
       }
 
       try {
-        Path dir = Paths.get(fileName).getParent();
+        final Path filePath = Paths.get(fileName);
+        final Path dir = filePath.getParent();
         if (!dirToFilesMap.containsKey(dir)) {
-          // There is not entry for this directory, create a watchkey
+          // There is no entry for this directory, create a watchkey
           WatchKey watchKey = dir.register(watchService,
               new WatchEvent.Kind[]{StandardWatchEventKinds.ENTRY_MODIFY},
-              SensitivityWatchEventModifier.HIGH);
+              SensitivityWatchEventModifier.LOW);
           keys.put(watchKey, dir);
         }
         // Add the config file to dir map
         dirToFilesMap.put(dir, fileName);
+        fileTimeMap.put(fileName, Files.getLastModifiedTime(filePath));
```

Comment:
Is this the full path name? It will be safer to use the full path, in case there are some configs that are in different directories that have the same file name.

Reply:
Yes, the file name is fully qualified.

Reply:
How about Path.getCanonicalPath(), or adding a comment saying that the key for the map should be the fully qualified path.